File: app/src/main/java/br/com/sailboat/zerotohero/view/ui/exercise/list/view_model/ExerciseListViewModel.java
package br.com.sailboat.zerotohero.view.ui.exercise.list.view_model;

import java.util.ArrayList;
import java.util.List;

import br.com.sailboat.zerotohero.base.BaseViewModel;
import br.com.sailboat.zerotohero.model.Exercise;

public class ExerciseListViewModel extends BaseViewModel {

    private final List<Exercise> exercises;

    public ExerciseListViewModel() {
        this.exercises = new ArrayList<>();
    }

    public List<Exercise> getExercises() {
        return exercises;
    }
}

File: androidApp/src/main/java/br/com/sailboat/zerotohero/android/feature/exercise/list/presentation/viewmodel/ExerciseListViewModel.kt
package br.com.sailboat.zerotohero.android.feature.exercise.list.presentation.viewmodel

import androidx.lifecycle.viewModelScope
import br.com.sailboat.zerotohero.android.base.BaseViewModel
import br.com.sailboat.zerotohero.domain.usecase.GetExercisesUseCase
import kotlinx.coroutines.launch

internal class ExerciseListViewModel(
    override val viewState: ExerciseListViewState = ExerciseListViewState(),
    private val getExercisesUseCase: GetExercisesUseCase,
) : BaseViewModel<ExerciseListViewState, ExerciseListViewAction>() {

    override fun dispatchViewAction(viewAction: ExerciseListViewAction) {
        when (viewAction) {
            ExerciseListViewAction.OnStart -> onStart()
        }
    }

    private fun onStart() = viewModelScope.launch {
        val exercises = getExercisesUseCase()
        viewState.exerciseList.value = exercises
    }
}
